### PR TITLE
Adding FMC support to STM32U5

### PR DESCRIPTION
**Pull request by potentialdiffer**
Adding fmc definition to device tree and solved a minor warning.

### CODE REVIEW

File: dts/arm/st/u5/stm32u5.dtsi [hunk 1]
```diff
@@ -149,6 +150,7 @@
 			compatible = "st,stm32-exti";
 			interrupt-controller;
 			#interrupt-cells = <1>;
+			#address-cells = <1>;
```
Comment (erwango) [inline dts/arm/st/u5/stm32u5.dtsi:153]: Thanks for fixing this, but please do it in a dedicated commit (mentioning the warning you fix).
Pushing a bit, even a dedicated PR would be nice so we can discuss more around this, as I'm not seeing this warning on my side, and maybe fix on all series.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for introducing this.

One request: would you mind enabling support on an in tree board (B-U585I-IOT02A), so it can be easily re-used by others and more importantly tested in CI.

File: dts/arm/st/u5/stm32u5.dtsi (hunk 1 above)
Comment (potentialdiffer) [reply dts/arm/st/u5/stm32u5.dtsi:153]: Sure, will create separate commits.

I can add the configuration for the B-U585I-IOT02A. I just saw, it has additional SRAM available. However, I will not be able to test it myself.

Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi:153]: > On the B-U585I-IOT02A, the SRAM is connected via SPI and not via FMC. The nucleo_u575zi_q also does not have any FMC connected hardware

Ok, then let it aside.

Comment (potentialdiffer) [reply dts/arm/st/u5/stm32u5.dtsi:153]: Alright, I removed the last commit. I think this request can be resolved now?

**Review by erwango — DISMISSED** [review]

File: dts/arm/st/u5/stm32u5.dtsi [hunk 2]
```diff
@@ -662,6 +664,20 @@
 			dma-offset = <0>;
 			status = "disabled";
 		};
+
+		fmc: memory-controller@420d0400 {
+			compatible = "st,stm32-fmc";
+			reg = <0x420d0400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB3 0x00001000>;
```
Comment (FRASTM) [inline dts/arm/st/u5/stm32u5.dtsi:671]: can you please confirm, I see the FSMC clock control on the  AHB2_2 bit 0   (AHB3 bit 12 is GTZC)

Comment (potentialdiffer) [reply dts/arm/st/u5/stm32u5.dtsi]: You are right, I double checked in the documentation. I have to say, I just took over the configuration from H7, which did not align with U5. I fixed the clocks definition.

File: dts/arm/st/u5/stm32u5.dtsi (hunk 1 above)
Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi:174]: Seems it reappeared somehow

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/u5/stm32u5.dtsi (hunk 2 above)
Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi]: One question then: did you test it ?

File: dts/arm/st/u5/stm32u5.dtsi [hunk 3]
```diff
@@ -683,6 +685,20 @@
 			dma-offset = <0>;
 			status = "disabled";
 		};
+
+		fmc: memory-controller@420d0400 {
+			compatible = "st,stm32-fmc";
+			reg = <0x420d0400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2_2 0x00000000>;
```
Comment (FRASTM) [inline dts/arm/st/u5/stm32u5.dtsi:692]:
```suggestion
			clocks = <&rcc STM32_CLOCK_BUS_AHB2_2 0x00000001>;
```

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: dts/arm/st/u5/stm32u5.dtsi (hunk 2 above)
Comment (potentialdiffer) [reply dts/arm/st/u5/stm32u5.dtsi]: Not yet, as the hardware is not yet ready. Will be able to test it next week.

File: dts/arm/st/u5/stm32u5.dtsi (hunk 3 above)
Comment (potentialdiffer) [reply dts/arm/st/u5/stm32u5.dtsi]: Fixed this

**Review by FRASTM — DISMISSED** [review]

File: dts/arm/st/u5/stm32u5.dtsi (hunk 2 above)
Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi]: ok. Then I prefer we don't merge and wait for tests results.
It's possible that FMC on U5 requires more than only dtsi addition. Some drivers changes might be required as well. This is usual on U5 series.

Comment (erwango) [reply dts/arm/st/u5/stm32u5.dtsi]: @potentialdiffer please add test configuration as part of this PR in a dedicated commit

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]
